extractor/filesystem/os/pacman/pacman.go

```diff
@@ -0,0 +1,273 @@
+		}
+		e.stats.AfterFileExtracted(e.Name(), &stats.FileExtractedStats{
+			Path:          input.Path,
+			Result:        filesystem.ExtractorErrorToFileExtractedResult(err),
+			FileSizeBytes: fileSizeBytes,
+		})
+	}
+	return inventory, err
+}
+
+func (e Extractor) extractFromInput(ctx context.Context, input *filesystem.ScanInput) ([]*extractor.Inventory, error) {
+	m, err := osrelease.GetOSRelease(input.FS)
+	if err != nil {
+		log.Errorf("osrelease.ParseOsRelease(): %v", err)
+	}
+
+	s := bufio.NewScanner(input.Reader)
+	var pkgName, pkgVersion, pkgDescription, pkgDependencies string
+	pkgs := []*extractor.Inventory{}
+
+	for s.Scan() {
+		line := s.Text()
+		line = strings.TrimSpace(line)
+
+		if len(line) == 0 {
+			continue
+		}
+
+		if strings.HasPrefix(line, "%NAME%") {
+			pkgName = extractValue(s)
+		} else if strings.HasPrefix(line, "%VERSION%") {
+			pkgVersion = extractValue(s)
+		} else if strings.HasPrefix(line, "%DESC%") {
+			pkgDescription = extractValue(s)
+		} else if strings.HasPrefix(line, "%DEPENDS%") {
+			pkgDependencies = extractValues(s)
+		}
+	}
+
+	i := &extractor.Inventory{
+		Name:    pkgName,
+		Version: pkgVersion,
+		Metadata: &Metadata{
+			PackageName:    pkgName,
+			PackageVersion: pkgVersion,
+			OSID:           m["ID"],
+			OSVersionID:    m["VERSION_ID"],
+		},
+		Locations: []string{input.Path},
+	}
+
+	if pkgDescription != "" {
+		i.Metadata.(*Metadata).PackageDescription = pkgDescription
+	}
+	if len(pkgDependencies) != 0 {
+
+		i.Metadata.(*Metadata).PackageDependencies = pkgDependencies
+	}
+
+	pkgs = append(pkgs, i)
+
+	return pkgs, nil
+}
+
+func extractValue(scanner *bufio.Scanner) string {
+	// Skip the current line.
+	scanner.Scan()
+
+	return strings.TrimSpace(scanner.Text())
+}
+
+func extractValues(scanner *bufio.Scanner) string {
+	var values []string
+
+	// Skip the current line.
+	scanner.Scan()
+
+	// Scan for multiple values.
+	for scanner.Scan() {
+		line := strings.TrimSpace(scanner.Text())
+
+		if len(line) == 0 || strings.HasPrefix(line, "%") {
+			break
+		}
+
+		values = append(values, line)
+	}
+
+	return strings.Join(values, ", ")
+}
+
+// Ecosystem returns the OSV Ecosystem of the software extracted by this extractor.
+func (Extractor) Ecosystem(i *extractor.Inventory) string {
+	m := i.Metadata.(*Metadata)
+	osID := cases.Title(language.English).String(toNamespace(m))
+	if m.OSVersionID == "" {
+		return osID
+	}
+	return osID + ":" + m.OSVersionID
+}
+
+func toNamespace(m *Metadata) string {
+	if m.OSID != "" {
+		return m.OSID
+	}
+	log.Errorf("os-release[ID] not set, fallback to 'arch'")
+	return "arch"
+}
+
+// ToPURL converts an inventory created by this extractor into a PURL.
+func (e Extractor) ToPURL(i *extractor.Inventory) *purl.PackageURL {
+	m := i.Metadata.(*Metadata)
+	q := map[string]string{}
+	distro := toDistro(m)
+	if distro != "" {
+		q[purl.Distro] = distro
+	}
+	if m.PackageDependencies != "" {
+		q[purl.PackageDependencies] = m.PackageDependencies
```

Comment:
While there's no official PURL spec for Arch Linux packages so we could come up with our own definition, I don't think we should include the package dependencies in it. We don't do that for dpkg, apk, etc. either.